### PR TITLE
add IntegrationTest

### PR DESCRIPTION
**Pull request by alesyalis**

### CODE REVIEW

**Review by ihhyy — APPROVED** [review]

File: Weather.СonsoleApp/Weather.IntegrationTest/Service/WeatherServiceIntegrationTest.cs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+using Weather.DataAccess.Repositories;
+using Weather.DataAccess.Repositories.Abstrdact;
+using Xunit;
+
+namespace Weather.IntegrationTest.Service
+{
+    public class WeatherServiceIntegrationTest
+    {
+        private WeatherService _weatherService;
+        private IWeatherRepository _weatherRepository;
```
Comment (Xodoc) [inline Weather.СonsoleApp/Weather.IntegrationTest/Service/WeatherServiceIntegrationTest.cs]: make fields readonly

File: Weather.СonsoleApp/Weather.IntegrationTest/Service/WeatherServiceIntegrationTest.cs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            Assert.Equal(message, messageResult);
+        }
+
+        [Fact]
+        public async Task GetWeatherAsync_ReceivedIncorrectWeather_IsErrorTrue()
+        {
+            // Arrange
+            var name = "gdrh";
+            _validator.ValidateCityByName(name);
+            await _weatherRepository.GetWeatherAsync(name);
```
Comment (Xodoc) [inline Weather.СonsoleApp/Weather.IntegrationTest/Service/WeatherServiceIntegrationTest.cs]: Remove

**Review by Xodoc — APPROVED** [review]

File: Weather.СonsoleApp/Weather.IntegrationTest/IConfigTest.cs [hunk 3]
```diff
@@ -0,0 +1,10 @@
+﻿using AppConfiguration.AppConfig;
+namespace Weather.IntegrationTest
+{
+    public class IConfigTest : IConfig
+    {
+        public string Url =>  "https://api.openweathermap.org/data/2.5/weather?q";
+
+        public string ApiKey => "8e943ed8b016561c73b8a1920366ef79";
+    }
```
Comment (denja2) [inline Weather.СonsoleApp/Weather.IntegrationTest/IConfigTest.cs]: Why class ia named from I?

File: Weather.СonsoleApp/Weather.IntegrationTest/IConfigTest.cs [hunk 4]
```diff
@@ -0,0 +1,10 @@
+﻿using AppConfiguration.AppConfig;
+namespace Weather.IntegrationTest
+{
+    public class IConfigTest : IConfig
+    {
+        public string Url =>  "https://api.openweathermap.org/data/2.5/weather?q";
```
Comment (denja2) [inline Weather.СonsoleApp/Weather.IntegrationTest/IConfigTest.cs]: Move to config

File: Weather.СonsoleApp/Weather.IntegrationTest/Service/WeatherServiceIntegrationTest.cs (hunk 1 above)
Comment (denja2) [reply Weather.СonsoleApp/Weather.IntegrationTest/Service/WeatherServiceIntegrationTest.cs]: +

File: Weather.СonsoleApp/Weather.IntegrationTest/Service/WeatherServiceIntegrationTest.cs [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+            _validator = new Validator();
+            _weatherService = new WeatherService(_weatherRepository, _validator);
+        }
+       
+        [Fact]
+        public async Task GetWeatherAsync_CorrectWeatherIsReceived_IsErrorFalseAndMessageIsGenerated()
+        {
+            // Arrange
+            var name = "Minsk";
+            _validator.ValidateCityByName(name);
```
Comment (denja2) [inline Weather.СonsoleApp/Weather.IntegrationTest/Service/WeatherServiceIntegrationTest.cs]: not needed

File: Weather.СonsoleApp/Weather.IntegrationTest/Service/WeatherServiceIntegrationTest.cs [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+            var model = await _weatherRepository.GetWeatherAsync(name);
+            var message = $"In {model.Name}: {model.Main.Temp} ";
+
+            //Act
+            var response = await _weatherService.GetWeatherAsync(name);
+            var messageResult = response.Message.Substring(0, response.Message.IndexOf('°'));
+            // Assert
+            Assert.False(response.IsError);
+            Assert.Equal(message, messageResult);
```
Comment (denja2) [inline Weather.СonsoleApp/Weather.IntegrationTest/Service/WeatherServiceIntegrationTest.cs]: Use regex

File: Weather.СonsoleApp/Weather.IntegrationTest/Service/WeatherServiceIntegrationTest.cs [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+            // Arrange
+            var name = "gdrh";
+            _validator.ValidateCityByName(name);
+            await _weatherRepository.GetWeatherAsync(name);
+
+            //Act
+            var response = await _weatherService.GetWeatherAsync(name);
+            
+            // Assert
+            Assert.True(response.IsError);
```
Comment (denja2) [inline Weather.СonsoleApp/Weather.IntegrationTest/Service/WeatherServiceIntegrationTest.cs:58]: what is the message?

**Review by denja2 — CHANGES_REQUESTED** [review]

File: Weather.СonsoleApp/Weather.IntegrationTest/Service/WeatherServiceIntegrationTest.cs (hunk 7 above)
Comment (alesyalis) [reply Weather.СonsoleApp/Weather.IntegrationTest/Service/WeatherServiceIntegrationTest.cs:58]: add

File: Weather.СonsoleApp/Weather.IntegrationTest/Service/WeatherServiceIntegrationTest.cs (hunk 5 above)
Comment (alesyalis) [reply Weather.СonsoleApp/Weather.IntegrationTest/Service/WeatherServiceIntegrationTest.cs]: done

File: Weather.СonsoleApp/Weather.IntegrationTest/IConfigTest.cs (hunk 3 above)
Comment (alesyalis) [reply Weather.СonsoleApp/Weather.IntegrationTest/IConfigTest.cs]: done

File: Weather.СonsoleApp/Weather.IntegrationTest/Service/WeatherServiceIntegrationTest.cs (hunk 1 above)
Comment (alesyalis) [reply Weather.СonsoleApp/Weather.IntegrationTest/Service/WeatherServiceIntegrationTest.cs]: done

File: Weather.СonsoleApp/Weather.IntegrationTest/Service/WeatherServiceIntegrationTest.cs (hunk 6 above)
Comment (alesyalis) [reply Weather.СonsoleApp/Weather.IntegrationTest/Service/WeatherServiceIntegrationTest.cs]: done

File: Weather.СonsoleApp/Weather.IntegrationTest/Service/WeatherServiceIntegrationTest.cs [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+using Xunit;
+
+namespace Weather.IntegrationTest.Service
+{
+    public class WeatherServiceIntegrationTest
+    {
+        private readonly WeatherService _weatherService;
+        private readonly IWeatherRepository _weatherRepository;
+        private readonly IValidator _validator;
+        private readonly ConfigTest _configuratio;
```
Comment (denja2) [inline Weather.СonsoleApp/Weather.IntegrationTest/Service/WeatherServiceIntegrationTest.cs]: name

File: Weather.СonsoleApp/Weather.IntegrationTest/Service/WeatherServiceIntegrationTest.cs [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+            _validator = new Validator();
+            _weatherService = new WeatherService(_weatherRepository, _validator);
+        }
+       
+        [Fact]
+        public async Task GetWeatherAsync_CorrectWeatherIsReceived_IsErrorFalseAndMessageIsGenerated()
+        {
+            // Arrange
+            var name = "Minsk";
+            var model = await _weatherRepository.GetWeatherAsync(name);
```
Comment (denja2) [inline Weather.СonsoleApp/Weather.IntegrationTest/Service/WeatherServiceIntegrationTest.cs]: ?

File: Weather.СonsoleApp/Weather.IntegrationTest/Service/WeatherServiceIntegrationTest.cs [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+            Assert.False(response.IsError);
+            Assert.Equal(message, messageResult[0]);
+        }
+
+        [Fact]
+        public async Task GetWeatherAsync_ReceivedIncorrectWeather_IsErrorTrue()
+        {
+            // Arrange
+            var name = "gdrh";
+            _validator.ValidateCityByName(name);
```
Comment (denja2) [inline Weather.СonsoleApp/Weather.IntegrationTest/Service/WeatherServiceIntegrationTest.cs]: remove

File: Weather.СonsoleApp/Weather.IntegrationTest/Service/WeatherServiceIntegrationTest.cs [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+            Assert.True(response.IsError);
+            Assert.Equal(message, response.Message);
+        }
+
+        [Fact]
+        public async Task GetWeatherAsync_ValidatorThrowsIsExeption_ReceivedError()
+        {
+            // Arrange
+            var name = "";
+            void actualResult() => _validator.ValidateCityByName(name);
```
Comment (denja2) [inline Weather.СonsoleApp/Weather.IntegrationTest/Service/WeatherServiceIntegrationTest.cs]: ?

File: Weather.СonsoleApp/Weather.IntegrationTest/Service/WeatherServiceIntegrationTest.cs [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+            await _weatherRepository.GetWeatherAsync(name);
+            var message = "Entering the city is required\n";
+
+            //Act
```
Comment (denja2) [inline Weather.СonsoleApp/Weather.IntegrationTest/Service/WeatherServiceIntegrationTest.cs:69]: call service in act

File: Weather.СonsoleApp/Weather.IntegrationTest/Service/WeatherServiceIntegrationTest.cs (hunk 2 above)
Comment (alesyalis) [reply Weather.СonsoleApp/Weather.IntegrationTest/Service/WeatherServiceIntegrationTest.cs]: done

File: Weather.СonsoleApp/Weather.IntegrationTest/IConfigTest.cs (hunk 4 above)
Comment (alesyalis) [reply Weather.СonsoleApp/Weather.IntegrationTest/IConfigTest.cs]: done

File: Weather.СonsoleApp/Weather.IntegrationTest/Service/WeatherServiceIntegrationTest.cs (hunk 8 above)
Comment (alesyalis) [reply Weather.СonsoleApp/Weather.IntegrationTest/Service/WeatherServiceIntegrationTest.cs]: done

File: Weather.СonsoleApp/Weather.IntegrationTest/Service/WeatherServiceIntegrationTest.cs (hunk 9 above)
Comment (alesyalis) [reply Weather.СonsoleApp/Weather.IntegrationTest/Service/WeatherServiceIntegrationTest.cs]: done

File: Weather.СonsoleApp/Weather.IntegrationTest/Service/WeatherServiceIntegrationTest.cs (hunk 10 above)
Comment (alesyalis) [reply Weather.СonsoleApp/Weather.IntegrationTest/Service/WeatherServiceIntegrationTest.cs]: done

File: Weather.СonsoleApp/Weather.IntegrationTest/Service/WeatherServiceIntegrationTest.cs (hunk 11 above)
Comment (alesyalis) [reply Weather.СonsoleApp/Weather.IntegrationTest/Service/WeatherServiceIntegrationTest.cs]: done

File: Weather.СonsoleApp/Weather.IntegrationTest/ConfigTest.cs [hunk 13]
```diff
@@ -0,0 +1,22 @@
...
+    {
+        public string Url { get; set; }
+
+        public string ApiKey { get; set; }
+
+        public ConfigTest()
+        {
+            var config = new ConfigurationBuilder().AddJsonFile("appsettings.json")
+                                                   .Build();
+            ApiKey = config["AppString:0:ApiKey"];
```
Comment (denja2) [inline Weather.СonsoleApp/Weather.IntegrationTest/ConfigTest.cs]: Refactor this

File: Weather.СonsoleApp/Weather.IntegrationTest/Service/WeatherServiceIntegrationTest.cs [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+        public async Task GetWeatherAsync_CorrectWeatherIsReceived_IsErrorFalseAndMessageIsGenerated()
+        {
+            // Arrange
+            var name = "Minsk";
+
+            //Act
+            var response = await _weatherService.GetWeatherAsync(name);
+            var regex = @"(\w?)(.*?)\.| (\B\W)(.*?)\.";
+            var description1 = "Dress warmly.";
+            var description2 = "It's fresh.";
```
Comment (denja2) [inline Weather.СonsoleApp/Weather.IntegrationTest/Service/WeatherServiceIntegrationTest.cs:34]: Regex in arrange

Comment (alesyalis) [reply Weather.СonsoleApp/Weather.IntegrationTest/Service/WeatherServiceIntegrationTest.cs:34]: done

File: Weather.СonsoleApp/Weather.IntegrationTest/ConfigTest.cs (hunk 13 above)
Comment (alesyalis) [reply Weather.СonsoleApp/Weather.IntegrationTest/ConfigTest.cs]: done

**Pull request merged**